src/navigation/actions.js

```diff
@@ -34,21 +34,37 @@ import { PREFERENCE_KEYS } from '../database/utilities/constants';
  *
  */
 
+/**
+ * Action creator for handling back navigation.
+ *
+ * Triggers back navigation and cleans up current state.
+ */
 export const goBack = () => dispatch => {
   if (!RootNavigator.canGoBack()) BackHandler.exitApp();
   else {
     UIDatabase.write(() => {
       const prescriptions = UIDatabase.objects('Prescription').filtered('status != "finalised"');
       UIDatabase.delete('Transaction', prescriptions);
 
-      batch(() => {
+      const prevRouteName = RootNavigator.getPrevRouteName();
+      const currRouteName = RootNavigator.getCurrentRouteName();
+
+      const navigateBack = () =>
         dispatch({
           ...NavigationActions.back(),
-          payload: {
-            prevRouteName: RootNavigator.getPrevRouteName(),
-          },
+          payload: { prevRouteName },
         });
+
+      const cleanUp = () => {
         dispatch(PrescriptionActions.deletePrescription());
+        if (FINALISABLE_PAGES.includes(currRouteName)) {
```

Comment:
I think you could just do this unconditionally, but no damage in a check right

Reply:
Yeah, think you're right, but no added benefit I can see... except for maybe reducing the branching complexity? But on the other hand dispatching unnecessary actions might make the code a bit less clear? On balance I think no harm in a little conditional :rofl:.